metrics-operator/controllers/common/providers/prometheus/common_test.go

```diff
@@ -185,8 +203,17 @@ func Test_GetRoundtripper(t *testing.T) {
 				t.Errorf("getRoundtripper() error = %s, wantErr %s", err.Error(), tt.errorStr)
 				return
 			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("getRoundtripper() got = %v, want %v", got, tt.want)
+			if !tt.wantErr && got == nil {
+				t.Errorf("getRoundtripper() returned nil, expected a RoundTripper")
+			}
+			// if !reflect.DeepEqual(got, tt.want) {
+			// 	t.Errorf("getRoundtripper() got = %v, want %v", got, tt.want)
+			// }
+			if tr, ok := got.(*http.Transport); ok {
```

Comment:
the insecureSkipVerify is not the only parameter that should be tested here, please test also the user and the password of the round tripper

Reply:
@odubajDT I have made the changes please verify it

Reply:
still only a single parameter is verified here, what about user/password and other parameters set in the struct?